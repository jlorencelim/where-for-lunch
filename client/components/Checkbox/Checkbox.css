.root {
  margin-left: 5px;
}
