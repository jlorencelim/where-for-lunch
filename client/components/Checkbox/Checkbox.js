import React from 'react';
import PropTypes from 'prop-types';
import styles from './Checkbox.css';

const Checkbox = ({ label, value, onChange }) => (
  <div className={styles.root}>
    <input
      type="checkbox"
      value={value}
      onChange={onChange}
    />
    <label>{label}</label>
  </div>
);

Checkbox.propTypes = {
  label: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  onChange: PropTypes.func,
  defaultChecked: PropTypes.bool,
};

export default Checkbox;
