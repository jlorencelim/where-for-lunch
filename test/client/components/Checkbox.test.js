import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from 'client/components/Checkbox/Checkbox';

test('With Jest snapshot, Button component renders with label and value', () => {
  const checkboxComponent = renderer
    .create(<Checkbox label="$" value={1} />)
    .toJSON();

  expect(checkboxComponent).toMatchSnapshot();
});
