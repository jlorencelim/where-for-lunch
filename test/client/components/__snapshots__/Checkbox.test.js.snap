// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest snapshot, Button component renders with label and value 1`] = `
<div
  className="root"
>
  <input
    type="checkbox"
    value={1}
  />
  <label>
    $
  </label>
</div>
`;
